From 85117890998f46030bdb67dc638e8adab5984c25 Mon Sep 17 00:00:00 2001
From: Moria Zuberi <moria.zuberi@tandemg.com>
Date: Thu, 12 May 2022 17:22:05 +0300
Subject: [PATCH] [SW-77385] erom: recovery: update load offset

Move erom-recovery image load offset to SRAM unused memory
instead of loading to SCRATCHPAD.
When loading to SCRATCHPAD we got memory overruns and erom-recovery
did not manage to flash OOB.

Change-Id: I222d1591e9131f5d01c19c5b954b8b9bdd03f12e
Signed-off-by: Moria Zuberi <moria.zuberi@tandemg.com>
---

diff --git a/arch/gaudi2/arc/dev_ext_rom.h b/arch/gaudi2/arc/dev_ext_rom.h
index 0baf793..6dcd79a 100644
--- a/arch/gaudi2/arc/dev_ext_rom.h
+++ b/arch/gaudi2/arc/dev_ext_rom.h
@@ -6,11 +6,13 @@
 #ifndef DEV_EXT_ROM_H_
 #define DEV_EXT_ROM_H_
 
+#define SDRAM_BASE			0x11000000
+
 #define EXT_ROM_START			(0x27FE2000)
 #define EXT_ROM_TEXT_LENGTH		(70 * 1024)
 /* No header bytes. Hence the text base is same as erom start address */
 #define EXT_ROM_PCIE_TEXT_BASE		(EXT_ROM_START)
-#define EXT_ROM_RECOVERY_TEXT_BASE	(EXT_ROM_START + 0x2000)
+#define EXT_ROM_RECOVERY_TEXT_BASE	(SDRAM_BASE + 0x10000)
 #define EXT_ROM_TPM_TEXT_BASE		(EXT_ROM_START)
 
 #endif
